src/main/kotlin/org/cascadebot/cascadebot/data/objects/TodoList.kt

```diff
@@ -79,41 +79,47 @@ class TodoList(val ownerId: Long) {
-        val buttonGroup = PersistentButtonGroup(memberId, channelId, guildId)
-        buttonGroup.addPersistentButton(PersistentButton.TODO_BUTTON_NAVIGATE_LEFT)
-        buttonGroup.addPersistentButton(PersistentButton.TODO_BUTTON_NAVIGATE_UP)
-        buttonGroup.addPersistentButton(PersistentButton.TODO_BUTTON_NAVIGATE_DOWN)
-        buttonGroup.addPersistentButton(PersistentButton.TODO_BUTTON_NAVIGATE_RIGHT)
-        buttonGroup.addPersistentButton(PersistentButton.TODO_BUTTON_CHECK)
-        return buttonGroup
+    private fun generateButtons(check: Boolean): ComponentContainer {
+        val container = ComponentContainer()
+        val row = CascadeActionRow()
+        if (check) {
+            row.addComponent(PersistentComponent.TODO_BUTTON_CHECK.component)
+        } else {
+            row.addComponent(PersistentComponent.TODO_BUTTON_UNCHECK.component)
+        }
+        row.addComponent(PersistentComponent.TODO_BUTTON_NAVIGATE_LEFT.component)
+        row.addComponent(PersistentComponent.TODO_BUTTON_NAVIGATE_UP.component)
+        row.addComponent(PersistentComponent.TODO_BUTTON_NAVIGATE_DOWN.component)
+        row.addComponent(PersistentComponent.TODO_BUTTON_NAVIGATE_RIGHT.component)
+        container.addRow(row)
+        return container
     }
 
     fun addUncheckButton(message: Message?) {
         val channel = CascadeBot.INS.client.getTextChannelById(channelId)
         if (channel != null) {
-            val data = GuildDataManager.getGuildData(channel.guild.idLong)
-            val buttonGroup = data.persistentButtons[channelId]!![messageId]
-            buttonGroup!!.addPersistentButton(PersistentButton.TODO_BUTTON_UNCHECK)
-            buttonGroup.removePersistentButton(PersistentButton.TODO_BUTTON_CHECK)
+            val container = GuildDataManager.getGuildData(channel.guild.idLong).componentCache[channelId]!![messageId]
+            container!!.getRow(0).setComponent(0, PersistentComponent.TODO_BUTTON_UNCHECK.component)
+            val data = GuildDataManager.getGuildData(message?.guild!!.idLong)
+            data.addComponents(channel, message, container)
         }
     }
 
     fun addCheckButton(message: Message?) {
         val channel = CascadeBot.INS.client.getTextChannelById(channelId)
         if (channel != null) {
-            val data = GuildDataManager.getGuildData(channel.guild.idLong)
-            val buttonGroup = data.persistentButtons[channelId]!![messageId]
-            buttonGroup!!.addPersistentButton(PersistentButton.TODO_BUTTON_CHECK)
-            buttonGroup.removePersistentButton(PersistentButton.TODO_BUTTON_UNCHECK)
+            val container = GuildDataManager.getGuildData(channel.guild.idLong).componentCache[channelId]!![messageId]
+            container!!.getRow(0).setComponent(0, PersistentComponent.TODO_BUTTON_CHECK.component)
+            val data = GuildDataManager.getGuildData(message?.guild!!.idLong)
+            data.addComponents(channel, message, container!!)
```

Comment:
These two could be extracted into 1 method that takes PersistentComponent as a arg